src/app/data-model/DecodableList.h

```diff
@@ -47,30 +109,13 @@ namespace DataModel {
  *
  */
 template <typename T>
-class DecodableList
+class DecodableList : public detail::DecodableListBase
 {
 public:
-    DecodableList() { ClearReader(); }
+    DecodableList() {}
 
     static constexpr bool kIsFabricScoped = DataModel::IsFabricScoped<T>::value;
 
-    /*
-     * @brief
-     *
-     * This call stores a TLV reader positioned on the list this class is to manage.
-     *
-     * Specifically, the passed-in reader should be pointing into the list just after
-     * having called `OpenContainer` on the list element.
-     */
-    void SetReader(const TLV::TLVReader & reader) { mReader = reader; }
-
-    /*
-     * @brief
-     *
-     * This call clears the TLV reader managed by this class, so it can be reused.
-     */
-    void ClearReader() { mReader.Init(nullptr, 0); }
-
     template <typename T0 = T, std::enable_if_t<DataModel::IsFabricScoped<T0>::value, bool> = true>
     void SetFabricIndex(FabricIndex fabricIndex)
     {
```

Comment:
Followup: we might get some more wins by moving bits of the iterator into a non-templated super-class too.

Reply:
created #36264 for followups